File: P1/fibo_recursivo.py
def fib(n):
    if n == 0:
        return 0
    if n == 1:
        return 1
    if n == 2:
        return 1
    sum = fib(n-1) + fib(n-2) + fib(n-3)
    return sum

#########################################################
## 1ra Pregunta Fibonnacci en programcacion recurisiva
## 0,1,1,2,4,7,13
#########################################################
n = int(input('Introduce un un número:\n>> '))
flag = -1
fibo=[]
while flag < n-1:
    flag += 1
    fibo.append(fib(flag))
print(fibo)

File: P1/fibo_recursivo.hs
fib::Int->Int->Int->Int->String
fib 0 a b c =""
fib n 0 0 0 = show(0)++" "++fib (n-1) 0 0 1
fib n 0 0 1 = show(1)++" "++fib (n-1) 0 1 1
fib n 0 1 1 = show(1)++" "++fib (n-1) 1 1 2
fib n a b c=show(c)++" "++fib (n-1) (b) (c) (a+b+c)

fibo::Int->String
fibo n= fib n 0 0 0

main :: IO ()
main = do
    n <- getLine
    let prn = fibo(read n)
    print prn